elements/lisk-tree/src/merkle_tree.ts

```diff
@@ -11,21 +11,63 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+/* eslint-disable no-bitwise */
+/* eslint-disable @typescript-eslint/prefer-for-of */
 
+import { hash } from '@liskhq/lisk-cryptography';
 import { Proof } from './types';
+import { EMPTY_HASH, LEAF_PREFIX, BRANCH_PREFIX } from './constant';
 
 export class MerkleTree {
-	// eslint-disable-next-line
 	private _root: Buffer;
-	// eslint-disable-next-line
-	public constructor(_initValues: Buffer[]) {
-		this._root = Buffer.alloc(0);
+	private _dataLength = 0;
+	private _data: { [key: string]: Buffer } = {};
+
+	public constructor(initValues: Buffer[] = []) {
+		this._dataLength = initValues.length;
+		if (this._dataLength === 0) {
+			this._root = EMPTY_HASH;
+			return;
+		}
+		const leafHashes = new Array<Buffer>(initValues.length);
+		for (let i = 0; i < initValues.length; i += 1) {
+			const value = Buffer.concat(
+				[LEAF_PREFIX, initValues[i]],
+				LEAF_PREFIX.length + initValues[i].length,
+			);
+			const key = hash(value);
+			leafHashes[i] = key;
+			this._data[key.toString('binary')] = value;
```

Comment:
I see that we are storing the data with the prefix included. Is there a specific reason to do this? I think is fine, but you store an extra byte and maybe is not necessary (not big deal anyway)